lms/services/canvas_studio.py

```diff
@@ -0,0 +1,221 @@
+            token_url,
+            self.redirect_uri(),
+            auth=(self._client_id, self._client_secret),
+            authorization_code=code,
+        )
+
+    def authorization_url(self, state: str) -> str:
+        """
+        Construct the authorization endpoint URL for Canvas Studio.
+
+        This constructs the authorization URL for the Canvas Studio instance
+        associated with the current application instance.
+
+        :param state: `state` query param for the authorization request
+        """
+        auth_url = urlunparse(
+            (
+                "https",
+                self._domain,
+                "api/public/oauth/authorize",
+                "",
+                urlencode(
+                    {
+                        "client_id": self._client_id,
+                        "response_type": "code",
+                        "redirect_uri": self.redirect_uri(),
+                        "state": state,
+                    }
+                ),
+                "",
+            )
+        )
+
+        return auth_url
+
+    def redirect_uri(self) -> str:
+        """Return OAuth redirect URI for Canvas Studio."""
+        return self._request.route_url("canvas_studio_api.oauth.callback")
+
+    def list_media_library(self) -> list[File]:
+        """
+        List the videos and collections for the current user.
+
+        The result of this call corresponds to what the user sees if they
+        visit Canvas Studio from within their LMS, or use the Canvas Studio
+        picker when creating a Page.
+        """
+
+        collections = self._api_request("v1/collections", CanvasStudioCollectionsSchema)
+        user_collection = None
+
+        files = []
+        for collection in collections:
+            if collection["type"] == "user":
+                user_collection = collection
+                continue
+
+            files.append(
+                {
+                    "type": "Folder",
+                    "display_name": collection["name"],
+                    "updated_at": collection["created_at"],
+                    "id": str(collection["id"]),
+                    "contents": {
+                        "path": self._request.route_url(
+                            "canvas_studio_api.collections.media.list",
+                            collection_id=collection["id"],
+                        )
+                    },
+                }
+            )
+
+        if user_collection:
+            files += self.list_collection(str(user_collection["id"]))
+
+        return files
+
+    def list_collection(self, collection_id: str) -> list[File]:
+        """List the videos in a collection."""
+
+        media = self._api_request(
+            f"v1/collections/{collection_id}/media", CanvasStudioCollectionMediaSchema
+        )
+
+        files = []
+        for item in media:
+            media_id = item["id"]
+            files.append(
+                {
+                    "type": "File",
```

Comment:
We should do a follow up PR to store these in the DB.

1. It's always useful for debugging 
2. It might be required for Course copy
3. Allows to retrieve that information for other features (ie edit functionality, but we don't do that yet).


Canvas example: https://github.com/hypothesis/lms/blob/moodle-error-handling/lms/services/canvas_api/client.py#L263

Reply:
OK, I'll put a note on the project board.